src/NBitcoin/NetworkBuilder.cs

```diff
@@ -1,141 +1,143 @@
-			}
-			SetConsensus(network.Consensus).
-			SetGenesis(network.GetGenesis()).
-			SetMagic(_Magic).
-			SetPort(network.DefaultPort).
-			SetRPCPort(network.RPCPort);
-			SetTxFees(network.MinTxFee, network.FallbackFee, network.MinRelayTxFee);
-		}
-
-		public NetworkBuilder AddAlias(string alias)
-		{
-			_Aliases.Add(alias);
-			return this;
-		}		
-
-		public NetworkBuilder SetRPCPort(int port)
-		{
-			_RPCPort = port;
-			return this;
-		}
-
-		public NetworkBuilder SetPort(int port)
-		{
-			_Port = port;
-			return this;
-		}
-
-
-		public NetworkBuilder SetMagic(uint magic)
-		{
-			_Magic = magic;
-			return this;
-		}
-
-#if !NOSOCKET
-		public NetworkBuilder AddDNSSeeds(IEnumerable<DNSSeedData> seeds)
-		{
-			vSeeds.AddRange(seeds);
-			return this;
-		}
-		public NetworkBuilder AddSeeds(IEnumerable<NetworkAddress> seeds)
-		{
-			vFixedSeeds.AddRange(seeds);
-			return this;
-		}
-#endif
-		
-		public NetworkBuilder SetConsensus(Consensus consensus)
-		{
-			_Consensus = consensus == null ? null : consensus.Clone();
-			return this;
-		}
-		
-		public NetworkBuilder SetGenesis(Block genesis)
-		{
-			_Genesis = genesis;
-			return this;
-		}		
-
-		public NetworkBuilder SetBase58Bytes(Base58Type type, byte[] bytes)
-		{
-			_Base58Prefixes.AddOrReplace(type, bytes);
-			return this;
-		}
-
-		public NetworkBuilder SetBech32(Bech32Type type, string humanReadablePart)
-		{
-			_Bech32Prefixes.AddOrReplace(type, Encoders.Bech32(humanReadablePart));
-			return this;
-		}
-		public NetworkBuilder SetBech32(Bech32Type type, Bech32Encoder encoder)
-		{
-			_Bech32Prefixes.AddOrReplace(type, encoder);
-			return this;
-		}
-
-		/// <summary>
-		/// Create an immutable Network instance, and register it globally so it is queriable through Network.GetNetwork(string name) and Network.GetNetworks().
-		/// </summary>
-		/// <returns></returns>
-		public Network BuildAndRegister()
-		{
-			return Network.Register(this);
-		}
-	}
+    public class NetworkBuilder
+    {
+        internal string Name;
+        internal Dictionary<Base58Type, byte[]> Base58Prefixes;
+        internal Dictionary<Bech32Type, Bech32Encoder> Bech32Prefixes;
+        internal List<string> Aliases;
+        internal int RPCPort;
+        internal int Port;
+        internal uint Magic;
+        internal Consensus Consensus;
+        internal List<DNSSeedData> Seeds ;
+        internal List<NetworkAddress> FixedSeeds;
+        internal Block Genesis;
+        internal long MinTxFee;
+        internal long FallbackFee;
+        internal long MinRelayTxFee;
+
+        public NetworkBuilder()
+        {
+            this.Base58Prefixes = new Dictionary<Base58Type, byte[]>();
+            this.Bech32Prefixes = new Dictionary<Bech32Type, Bech32Encoder>();
+            this.Aliases = new List<string>();
+            this.Seeds = new List<DNSSeedData>();
+            this.FixedSeeds = new List<NetworkAddress>();
+        }
+
+        public NetworkBuilder SetName(string name)
```

Comment:
very Java looking. Do we need this?
I see now that we use this construct so that we can chain the construction of NetworkBuilder.
It doesn't seem to give any advantage over a normal object construction.
Can we get rid of that and just use properties instead of internal fields?

Reply:
Method chaining? I find it useful.